pkg/yang/modules_test.go

```diff
@@ -130,6 +132,247 @@ func TestModulesFindByNamespace(t *testing.T) {
 	}
 }
 
+func TestModuleLinkage(t *testing.T) {
+	tests := []struct {
+		desc          string
+		inMods        map[string]string
+		wantErrSubstr string
+	}{{
+		desc: "invalid import",
+		inMods: map[string]string{
+			"dev": `
+				module dev {
+					prefix d;
+					namespace "urn:d";
+					import sys { prefix sys; }
+
+					revision 01-01-01 { description "the start of time"; }
+
+					deviation /sys:sys/sys:hostname {
+						deviate not-supported;
+					}
+				}`,
+			"sysdb": `
+				module sysdb {
+					prefix s;
+					namespace "urn:s";
+
+					revision 01-01-01 { description "the start of time"; }
+
+					container sys { leaf hostname { type string; } }
+				}`,
+		},
+		wantErrSubstr: "no such module",
```

Comment:
Perhaps remove the second module here -- it took me a moment to spot that `sys` != `sysdb` in the imports.

Reply:
Done.